### PR TITLE
Adding a fade in animation with a delay of ~200ms for table loading [UI-53]

### PR DESCRIPTION
**Pull request by teamchong**
Loading loading text in table loading.

- Adding a fade in animation with a delay of ~200ms.
- Avoid loading text overflow when table cell is not wide enough

### CODE REVIEW

**Comment by teamchong** [comment]
Add a opacity animations to the loading texts
2024-06-21 at 9 18 57 - Beige Parrot
2024-06-21 at 9 23 27 - Red Panda
2024-06-21 at 9 25 23 - Sapphire Python

File: packages/components/src/components/tables/es-table/es-table.css [hunk 1]
```diff
@@ -10,3 +10,18 @@ es-table {
...
+}
+
+@keyframes fadeIn {
+    from {
+        opacity: 0;
+    }
+    to {
+        opacity: 1;
+    }
+}
```
Comment (George-Payne) [inline packages/components/src/components/tables/es-table/es-table.css:27]: Let's move this up to `Page`, on it's `main`. Adding it only to the table means that we will still get a flash of the page title, etc, and also means it needs to be added in multiple places, especially if we were to add the loading to non-table based screens.

File: packages/components/src/components/es-loading-text/es-loading-text.css [hunk 2]
```diff
@@ -6,4 +6,18 @@
...
+@keyframes shimmer {
+    0% {
+        opacity: 0.5;
+    }
+    50% {
+        opacity: 1;
+    }
+    100% {
+        opacity: 0.5;
+    }
```
Comment (George-Payne) [inline packages/components/src/components/es-loading-text/es-loading-text.css:22]: I think if we are to add this, it should be more subtle. I'm not too keen on multiple things animating, especially in unison. I think for now, lets just leave it static (aside from the initial fade in) and possibly revisit it in the future.

File: packages/components/src/components/es-loading-text/es-loading-text.css [hunk 3]
```diff
@@ -6,4 +6,18 @@
     white-space: pre;
     color: transparent;
     user-select: none;
+    max-width: 100%;
```
Comment (George-Payne) [inline packages/components/src/components/es-loading-text/es-loading-text.css:9]: Does this solve the square end clipping issue?

**Review by George-Payne — CHANGES_REQUESTED** [review]

File: packages/components/src/components/es-loading-text/es-loading-text.css (hunk 3 above)
Comment (teamchong) [reply packages/components/src/components/es-loading-text/es-loading-text.css:9]: 2024-06-24 at 13 29 36 - Emerald Cicada

Comment (teamchong) [reply packages/components/src/components/es-loading-text/es-loading-text.css:9]: 2024-06-24 at 13 45 16 - Amaranth Tahr

File: packages/components/src/components/tables/es-table/es-table.css (hunk 1 above)
Comment (teamchong) [reply packages/components/src/components/tables/es-table/es-table.css:27]: Let me figure out how to share the table `loading` state to the Page component effectively. I’ll explore the following approaches:

1. Event-Driven State Management:
   - Emit custom events to notify the parent component or document.body about loading state changes.
   - Aggregate these states in a centralized place to determine the overall loading status.
2. Global State Management:
   - Use a simple global state management solution to track the loading state.
   - Make sure the parent page element reacts to changes in this global state.

Comment (George-Payne) [reply packages/components/src/components/tables/es-table/es-table.css:27]: `Page` already has the page state (including loading).

https://github.com/EventStore/Design-System/blob/main/packages/layout/src/components/Page/Page.tsx#L140

```diff
export const Page: FunctionalComponent<PageProps> = (props, children) => (
    <>
        <PageTitle>{props.pageTitle}</PageTitle>
        <main 
            ref={(ref) => ((document as any).main = ref)}
+           class={{ loading: props.state === "loading" }} 
        >
            {props.crumbs && <es-breadcrumb crumbs={props.crumbs} />}
            <PageBody {...props}>{children}</PageBody>
        </main>
        <PageView title={props.pageTitle} />
    </>
);
```

https://github.com/EventStore/Design-System/blob/main/packages/layout/src/css/page.css#L25

```diff
    height: 100%;
    padding: 30px 40px;
+
+    &.loading {
+        opacity: 0;
+        animation: fadeIn 2s forwards;
+        animation-delay: 0.2s;
+    }
}

+@keyframes fadeIn {
+    from {
+        opacity: 0;
+    }
+    to {
+        opacity: 1;
+    }
+}
```

Comment (teamchong) [reply packages/components/src/components/tables/es-table/es-table.css]: 2024-06-24 at 17 38 59 - Gold Macaw

Comment (George-Payne) [reply packages/components/src/components/tables/es-table/es-table.css]: Cool, looking at the timings in the cloud UI, I think we could bump the delay up to `400` and reduce the fade to our standard `500`. 

We also need to set the duration to `0` for `prefers-reduced-motion` users.

Comment (teamchong) [reply packages/components/src/components/tables/es-table/es-table.css]: 2024-06-24 at 23 39 57 - Gray Falcon

Comment (teamchong) [reply packages/components/src/components/tables/es-table/es-table.css]: 2024-06-24 at 23 44 18 - Indigo Peafowl

**Review by George-Payne — APPROVED** [review]

**Pull request merged**